source/codegen/validate_examples.py

```diff
@@ -40,18 +40,19 @@ def validate_examples(driver_glob_expression: str, ip_address: str, device_name:
         )
         for dir in examples_dir.glob(driver_glob_expression):
             print()
-            print(f"-- Validating: {dir.name} --")
+            print(f"-- Validating: {dir.name}")
 
             print(f" -> Running black line-length 100")
             system(f"poetry run black --check --line-length 100 {dir}")
 
             print(f" -> Running mypy")
-            system(f"poetry run mypy {dir}")
+            system(f"poetry run mypy {dir} --check-untyped-defs")
 
-            for file in dir.glob("*.py"):
-                print(f" -> Running example: {file.name}")
-                PORT = 31763
-                system(rf"poetry run python {file} {ip_address} {PORT} {device_name}")
+            if ip_address:
```

Comment:
We should add a argument to run simulation too. makes sense to not run them for build right now, but in general a workflow that can easily test simulated (by not passing in the ip_address).

Reply:
We run simulated by not passing in a `device_name` (the examples will use simulated when there's no 3rd argument). Running on simulated still requires that there be a server running (with drivers installed if you want the example to pass), so you still need the `ip_address`. It just means that the server doesn't need to have any physical hardware.